packages/test/test-end-to-end-tests/src/test/entryPointCompat.spec.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { describeLoaderCompat, describeNoCompat } from "@fluid-internal/test-version-utils";
+import {
+	ContainerRuntimeFactoryWithDefaultDataStore,
+	DataObject,
+	DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { FluidObject } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+
+describe("entryPoint compat", () => {
+	let provider: ITestObjectProvider;
+
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	async function getDefaultFluidObject(runtime: IContainerRuntime): Promise<FluidObject> {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		return (await runtime.getAliasedDataStoreEntryPoint?.("default"))!.get();
+	}
+
+	async function createRequestOnlyContainer(): Promise<IContainer> {
+		const dataObjectFactory = new DataObjectFactory("TestDataObject", TestDataObject, [], []);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(dataObjectFactory, [
+			[dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+		]);
+
+		return provider.createContainer(runtimeFactory);
+	}
+
+	async function createContainer(): Promise<IContainer> {
+		const dataObjectFactory = new DataObjectFactory("TestDataObject", TestDataObject, [], []);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataObjectFactory,
+			[[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+			undefined,
+			undefined,
+			undefined,
+			async (runtime: IContainerRuntime) => getDefaultFluidObject(runtime),
+		);
+
+		return provider.createContainer(runtimeFactory);
+	}
+
+	describeNoCompat("no compat", (getTestObjectProvider) => {
+		beforeEach(async () => {
+			provider = getTestObjectProvider();
+		});
+
+		it("entryPoint pattern", async () => {
+			const container = await createContainer();
+			const entryPoint = await container.getEntryPoint?.();
+			assert.notStrictEqual(entryPoint, undefined, "entryPoint was undefined");
+		});
+
+		it("request pattern", async () => {
+			const container = await createRequestOnlyContainer();
+			const requestResult = await container.request({ url: "/" });
+
+			assert.strictEqual(requestResult.status, 200, requestResult.value);
```

Comment:
You can also assert that `entryPoint` is here, right?

Reply:
This test just verifies request pattern still works (we don't care to check entryPoint in this test).